core/remote/watcher.js

```diff
@@ -416,7 +427,22 @@ class RemoteWatcher {
         break
       case 'DirMove':
         log.info({path, oldpath: change.was.path}, 'folder was moved or renamed remotely')
-        await this.prep.moveFolderAsync(sideName, change.doc, change.was)
+        const needUpdates /*: any */ = []
+        const newRemoteRevs /*: RemoteRevisionsByID */ = {}
+        for (let descendant of (change.descendantMoves || [])) {
+          if (descendant.update) {
+            needUpdates.push(descendant)
+          }
+          if (descendant.doc.remote) {
+            // $FlowFixMe
+            newRemoteRevs[descendant.doc.remote._id] = descendant.doc.remote._rev
+          }
+        }
+        await this.prep.moveFolderAsync(sideName, change.doc, change.was, newRemoteRevs)
+        for (let needUpdate of needUpdates) {
+          change.was = await this.pouch.byRemoteIdMaybeAsync(change.was.remote._id)
```

Comment:
I'm always uncomfortable with the *needUpdate(s)* logic here an in a few other places...
In this case, can't we just assume that child moves with updates need a refetch in Merge, where we're already iterating on them?
Also this makes me believe even more that we should passe data structures representing the changes to Prep/Merge, not the final doc/was which we end up fixing everywhere...

Reply:
Also I believe it's called `needRefetch` in other places, not `needUpdate`.

Reply:
Agree, but not part of this PR

Reply:
`needRefetch` is not the same as `update`

Reply:
Yep, but isn't `needUpdate` used exactly in the same way as `needRefetch` in `LocalWatcher`, aka reloading `old` / `was` from `Pouch` because some previously merged change of the same batch probably changed it?

Reply:
cleaned up